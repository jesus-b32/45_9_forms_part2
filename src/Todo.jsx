import React from "react";

function Todo ({id, task, handleRemove}) {
    // handle removing a todo when click event happens
    const remove = () => handleRemove(id);

    return (
        <div>
            <div>
                {task}
            </div>
            <button onClick={remove}>Remove Todo</button>
        </div>
    )
}

export default Todo;