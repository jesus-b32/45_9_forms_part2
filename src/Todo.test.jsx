import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';
import { describe, it, expect, vi } from 'vitest';

describe('Todo', () => {
    //smoke test
    it('renders App component', () => {
        render(<Todo/>);
    })

    //snapshot test
    it('matches snapshot', () => {
        const {asFragment} = render(<Todo/>);
        expect(asFragment()).toMatchSnapshot();
    })

    it("runs the delete function on button click", function() {
        const removeMock = vi.fn();
        const { getByText } = render(<Todo handleRemove={removeMock} />);
        const deleteButton = getByText("Remove Todo");
        fireEvent.click(deleteButton);
        expect(removeMock).toHaveBeenCalled();
    });
})