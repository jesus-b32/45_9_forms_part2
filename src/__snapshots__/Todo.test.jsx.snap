// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
    <button>
      Remove Todo
    </button>
  </div>
</DocumentFragment>
`;

exports[`Todo > matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
    <button>
      Remove Todo
    </button>
  </div>
</DocumentFragment>
`;
